plugins/rest/aws.go

```diff
@@ -318,6 +318,150 @@ func (cs *awsMetadataCredentialService) credentials(ctx context.Context) (aws.Cr
 	return cs.creds, nil
 }
 
+// awsCustomIdentityCredentialService represents an STS credential service that uses active IAM credentials
+// to obtain temporary security credentials generated by AWS STS for allowing existing IAM users access to AWS resources
+// that they don't already have access to.
+type awsCustomIdentityCredentialService struct {
+	RoleArn          string
+	AccessKey        string
+	SecretKey        string
+	SessionToken     string
+	RegionName       string `json:"aws_region"`
+	SessionName      string `json:"session_name"`
+	Domain           string `json:"aws_domain"`
+	stsURL           string
+	creds            aws.Credentials
+	expiration       time.Time
+	logger           logging.Logger
+	signatureVersion string
+}
+
+func (cs *awsCustomIdentityCredentialService) populateFromEnv() error {
+	cs.RoleArn = os.Getenv(awsRoleArnEnvVar)
+	if cs.RoleArn == "" {
+		return errors.New("no " + awsRoleArnEnvVar + " set in environment")
+	}
+
+	cs.AccessKey = os.Getenv(accessKeyEnvVar)
+	if cs.AccessKey == "" {
+		return errors.New("no " + accessKeyEnvVar + " set in environment")
+	}
+	cs.SecretKey = os.Getenv(secretKeyEnvVar)
+	if cs.SecretKey == "" {
+		return errors.New("no " + secretKeyEnvVar + " set in environment")
+	}
+
+	cs.SessionToken = os.Getenv(sessionTokenEnvVar)
+	if cs.SessionToken == "" {
+		// In case of missing SessionToken try to get SecurityToken
+		// AWS switched to use SessionToken, but SecurityToken was left for backward compatibility
+		cs.SessionToken = os.Getenv(securityTokenEnvVar)
```

Comment:
`AWS_SESSION_TOKEN`/`AWS_SECURITY_TOKEN` is optional?

Reply:
I am looking to test this out as part of an integration test. I'll update when I have some verification.

Reply:
I did an integration test using [this](https://docs.localstack.cloud/user-guide/aws/sts/) and did not have to provide `AWS_SESSION_TOKEN `. Also the end-to-end flow was successful.